neard/src/cli.rs

```diff
@@ -817,6 +818,56 @@ impl ValidateConfigCommand {
     }
 }
 
+fn normalize_whitespace(s: &str) -> String {
+    s.split_whitespace().collect::<Vec<_>>().join(" ")
+}
+
+/// Checks the provided kernel parameter  from /proc/sys
+/// and prints an error if it is not set to the expected value.
+fn check_kernel_param(param_name: &str, expected_value: &str) {
+    // Convert the dotted param_name into a path under /proc/sys
+    // For example, "net.core.rmem_max" -> "/proc/sys/net/core/rmem_max"
+    let mut path = PathBuf::from("/proc/sys");
+    for part in param_name.split('.') {
+        path.push(part);
+    }
+
+    match read_to_string(&path) {
+        Ok(contents) => {
+            let actual = contents.trim();
+            let actual_normalized = normalize_whitespace(actual);
+            let expected_normalized = normalize_whitespace(expected_value);
+
+            if actual_normalized != expected_normalized {
+                error!(
+                    "ERROR: {} is set to {}, expected {}",
+                    param_name, actual_normalized, expected_normalized
+                );
+            } else {
+                info!("OK: {} is set to expected value {}", param_name, expected_normalized);
+            }
+        }
+        Err(e) => {
+            error!("ERROR: failed to read parameter {} from {}: {}", param_name, path.display(), e);
+        }
+    }
+}
+
+/// Checks if the system has the expected values for the sysctl parameters.
+fn check_kernel_params() {
```

Comment:
Can we gurad this to only work on linux?
```suggestion
#[cfg(not(target_os = "linux"))]
fn check_kernel_params() {}

#[cfg(target_os = "linux")]
fn check_kernel_params() {
```

Reply:
I guess you meant #[cfg(target_os = "linux")]